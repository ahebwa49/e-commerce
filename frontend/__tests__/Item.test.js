import ItemComponent from "../components/Item";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

const fakeItem = {
  id: "12345",
  title: "A cool item",
  description: "This item is really cool",
  price: 4000,
  image: "dog.jpg",
  largeImage: "largeImage.jpg"
};

describe("<Item />", () => {
  it("renders and matches the matches the snapshot", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  /*it("renders and displays the title and pricetag", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = wrapper.find("PriceTag");
    //console.log(PriceTag.children().text());
    //console.log(wrapper.debug());
    expect(PriceTag.children().text()).toBe("$50");
    expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
  });
  it("renders the image properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const img = wrapper.find("img");
    //console.log(img.debug());
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });
  it("renders out the buttons properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    //console.log(wrapper.debug());
    const buttonList = wrapper.find(".buttonList");
    console.log(buttonList.debug());
    console.log(buttonList.children().length);
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find("Link").exists()).toBe(true);
    expect(buttonList.find("Link")).toHaveLength(1);
    expect(buttonList.find("Link")).toBeTruthy();
    expect(buttonList.find("AddToCart").exists()).toBe(true);
    expect(buttonList.find("DeleteItem").exists()).toBe(true);
  });*/
});
