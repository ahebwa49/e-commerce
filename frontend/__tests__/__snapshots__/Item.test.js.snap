// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "12345",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    This item is really cool
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "12345",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="12345"
    />
    <DeleteItem
      id="12345"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
